File: editorial/chap6-16_20/prob6-16.c
#include <stdio.h>

int main() {
	// 入力
	int N, X;
	scanf("%d%d", &N, &X);

	// すべての (a, b, c) の組み合わせを試す
	int answer = 0;
	int a, b, c;
	for (a = 1; a <= N; a++) {
		for (b = a + 1; b <= N; b++) {
			for (c = b + 1; c <= N; c++) {
				if (a + b + c == X) {
					answer += 1;
				}
			}
		}
	}

	// 答えを出力
	printf("%d\n", answer);

	return 0;
}

File: editorial/chap6-16_20/prob6-16.py
# 入力
N, X = map(int, input().split())

# すべての (a, b, c) の組み合わせを試す
answer = 0
for a in range(1, N + 1):
	for b in range(a + 1, N + 1):
		for c in range(b + 1, N + 1):
			if a + b + c == X:
				answer += 1

# 答えの出力
print(answer)